### ISSUE TITLE
poolformer_m36 correctness regression

### ISSUE THREAD

**Issue by desertfire** [pytorch/torchdynamo#1861]
This is actually a regression from https://github.com/pytorch/pytorch/pull/87054

Repro:
```
python benchmarks/dynamo/timm_models.py --training --accuracy --device cuda --inductor --float32 --only poolformer_m36
```
Error:
```
[2022-11-14 16:28:33,509] torch._dynamo.utils: [ERROR] RMSE (res-fp64): nan, (ref-fp64): 0.00001 and shape=torch.Size([96, 3, 7, 7])
[2022-11-14 16:28:33,509] torch._dynamo.utils: [ERROR] Accuracy failed for key name patch_embed.proj.weight.grad
```

**Comment by ngimel** [comment]
Hopefully will be fixed by #88927, but it's also flaky, even before #88927 I couldn't repro locally.

**Comment by desertfire** [comment]
Confirmed #88927 fixes this!
